libraries/chain/controller.cpp

```diff
@@ -2780,10 +2780,10 @@ struct controller_impl {
             const auto& bsp = std::get<std::decay_t<decltype(forkdb.chain_head)>>(cb.bsp);
 
             if( s == controller::block_status::incomplete ) {
-               forkdb.add( bsp );
-               forkdb.mark_valid( bsp );
+               const bool mark_valid = true;
+               const bool ignore_duplicate = false;
+               forkdb.add( bsp, mark_valid, ignore_duplicate );
```

Comment:
I'd like it even better if fork_database exported the types:

```
enum class mark_valid{ off, on };
enum class ignore_duplicate { off, on };
```

The signature of add was:
`add(const BSP& next_block, mark_valid do_mark_valid, ignore_duplicate do_ignore_duplicate);`

and it was called as:

` forkdb.add( bsp, mark_valid::on, ignore_duplicate::off);`

Not only it is clearer, but you cannot pass the parameters in the wrong order by mistake.

Reply:
Much better. Seems like our canonical use is `{ no, yes }`

Reply:
Added:
```cpp
   enum class mark_valid_t { no, yes };
   enum class ignore_duplicate_t { no, yes };
```